Sources/NIOSOCKS/Messages/Messages.swift

```diff
@@ -0,0 +1,66 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2021 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+/// Sent by the client and received by the server.
+public enum ClientMessage: Hashable {
+    
+    /// Contains the proposed authentication methods.
+    case greeting(ClientGreeting)
+    
+    /// Instructs the server of the target host, and the type of connection.
+    case request(SOCKSRequest)
+    
+    /// Data that is sent once the handshake is complete. Handling this case
+    /// should be rare, as you should remove the server handler after the handshake
+    /// process is complete. It can also be used when authenticating in response
+    /// to server challenges.
```

Comment:
This latter comment is not quite right: it's used by the server to process unknown auth message types.